File: prob23.js
// https://projecteuler.net/problem=22
// Completed 3/28/2022

// Finding abundant numbers
let abundantNumbers = [];
for(let i = 0; i <= 28123; i++){
    let divSum = 0;
    for(let j = 1; j <= i/2; j++){
        if(i % j === 0){
            divSum += j;
        }
    }
    if(divSum > i){
        abundantNumbers.push(i);
    }
}

// Finding sums of abundants
let abundantSums = [];
for(let i = 0; i < abundantNumbers.length; i++){
    for(let j = 0; j < abundantNumbers.length; j++){
        abundantSums.push(abundantNumbers[i] + abundantNumbers[j]);
    }
}

// Finding integers that are not in the abundantSums list
let cannotSum = 0;
for(let i = 0; i <= 28123; i++){
    let validation = true;
    for(let j = 0; j < abundantSums.length; j++){
        if(i === abundantSums[j]){
            validation = false;
            break;
        }
    }
    if(validation === true){
        cannotSum += i;
    }
}

console.log(cannotSum);

File: prob23.py
# https://projecteuler.net/problem=22
# Completed 3/28/2022

import math

def isAbundant(num):
    divSum = 0
    for i in range(1, math.floor(num/2)+1):
        if num % i == 0:
            divSum += i
    if divSum > num:
        return True
    return False

def lessThan28124(num):
    return num < 28124

abundants = list(filter(isAbundant, range(28124)))
abundantSums = []
sum = 0;

for i in abundants:
    for j in abundants:
        abundantSums.append(i + j)

abundantSums.sort()
abundantSums = list(dict.fromkeys(abundantSums))
abundantSums = list(filter(lessThan28124, abundantSums))

for i in range(28124):
    if i not in abundantSums:
        sum += i

print(sum)